packages/analytics/amplify_analytics_pinpoint/lib/src/flutter_path_provider/flutter_path_provider.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Multi platform class that provides storage location path
+/// Requires Flutter specific dependencies
+export 'flutter_path_provider_none.dart'
+    if (dart.library.html) 'flutter_path_provider_html.dart'
+    if (dart.library.io) 'flutter_path_provider_io.dart';
```

Comment:
```suggestion
export 'flutter_path_provider_none.dart'
    if (dart.library.io) 'flutter_path_provider_io.dart';
```

You could return the stub on web given that it throws unimplemented anyway.

Reply:
I throw different exceptions depending on the platform.  I think the difference can be helpful as the context in which the exception could be thrown is different.  In one case the user is trying to use a platform that just doesn't work, in the other case an implementation is wrong and the method is being called on web when it shouldn't be. 

throw UnsupportedError('Do not call FlutterPathProvider on web');

vs

throw UnsupportedError(
      'No suitable implementation was found on this platform.',
    );